app/questionnaire/questionnaire_store_updater.py

```diff
@@ -158,19 +158,51 @@ def remove_list_item_and_answers(self, list_name: str, list_item_id: str):
     def get_relationship_answers_for_list_name(
         self, list_name: str
     ) -> Union[List[Answer], None]:
-        assosciated_relationship_collectors = (
+        associated_relationship_collectors = (
             self._get_relationship_collectors_by_list_name(list_name)
         )
-        if not assosciated_relationship_collectors:
+        if not associated_relationship_collectors:
             return None
 
         relationship_answer_ids = [
             self._schema.get_relationship_answer_id_for_block(block["id"])
-            for block in assosciated_relationship_collectors
+            for block in associated_relationship_collectors
         ]
 
         return self._answer_store.get_answers_by_answer_id(relationship_answer_ids)
 
+    def update_same_name_items(self, list_name: str, same_name_answer_ids: List[str]):
+        same_name_items = set()
+        people_names: Dict[str, list] = {}
+
+        if not same_name_answer_ids:
+            return
+
+        list_model = self._questionnaire_store.list_store[list_name]
+
+        for current_list_item_id in list_model:
+            answers = self._questionnaire_store.answer_store.get_answers_by_answer_id(
+                answer_ids=same_name_answer_ids, list_item_id=current_list_item_id
+            )
+            current_names = [str(answer.value).lower() for answer in answers if answer]
+            current_list_item_name = " ".join(current_names)
+
+            if current_list_item_name in people_names:
+                people_names[current_list_item_name] += [current_list_item_id]
+            else:
+                people_names[current_list_item_name] = [current_list_item_id]
```

Comment:
The If else could be avoided by using a `defaultdict`

Reply:
This and the comment above were on outdated commits, because I failed to commit earlier in the day.